Remove unused language resources

raguiar9080 opened the pull request:
So as to try and keep the amount of space of our builds to a minimum. Lingua uses a lot of space if we have all languages present.
This current build is 25mb while the previous build was over 150mb with all dependencies included.
Since Lingua is only to be used on migration of the projects to aid message templates, it should be dropped in the future. Hence, generating the Jar with the deleted language resources and having it as an unmanaged dependency for now on the engine should prove sufficient. It will avoid us having to deal with package deploy in Kotlin (currently working locally).
Again, this will cut build space by a very large margin (I believe building the engine with the original would add almost 300mb instead of the now 25mb).

Signed-off-by: Ruben Aguiar <ruben@triggerise.org>

Commit:
Remove unused language resources

So as to try and keep the amount of space of our builds to a minimum.
Lingua uses a lot of space if we have all languages present.
This current build is under 30mb while the previous build was over 150mb
with all dependencies included.

Signed-off-by: Ruben Aguiar <ruben@triggerise.org>

pemistahl commented:
Hello Rúben,

I just saw your obviously accidental pull request in my repository. Can I help you somehow? I don't quite understand what you are trying to achieve. You want to reduce disk space by removing all language models from the library? Are you aware that Lingua will not work anymore after doing this? The rule engine might still work but it will certainly crash when it tries to access the language models. Even if the library does not crash (what I seriously doubt), the detection accuracy will drop to nearly 0%.

Feel free to open an issue in my repository if you can need some help. Thanks.

raguiar9080 commented:
> Hello Rúben,
> 
> I just saw your obviously accidental pull request in my repository. Can I help you somehow? I don't quite understand what you are trying to achieve. You want to reduce disk space by removing all language models from the library? Are you aware that Lingua will not work anymore after doing this? The rule engine might still work but it will certainly crash when it tries to access the language models. Even if the library does not crash (what I seriously doubt), the detection accuracy will drop to nearly 0%.
> 
> Feel free to open an issue in my repository if you can need some help. Thanks.

Hello Peter,

First of all, apologies for spamming you, I erratically opened the PR with the wrong destination. Thank for your offer to help, it's truly appreciated.
As you guessed, I was trying to make some 'hackish' attempt to remove some languages and then test it to see if it would work 🙈 . The main idea was to remove unused languages in lingua since for our purposes we only need to detect 5 languages. The purpose is to reduce the space of our builds since we're starting to see some issues due to this.
Is this not possible at all or it would require more changes in the engine side to be working? I expected the engine to definitely crash if I tried to access the removed languages but hoped for the non removed ones to continue to work. At this point I haven't fully tested it.
Since this is a very particular use case tied to our use of languages I felt opening a PR on your side wouldn't really help anyone but ourselves hence our fork.

Thanks in advance for the help

pemistahl commented:
> Is this not possible at all or would it require more changes in the engine side to be working?

No, this is currently not possible. I have already thought about making each language a separate dependency so that people can decide themselves which languages they want to add. I'm just not sure yet how to approach this because I don't want to break existing code of the users of my library. Nevertheless, I will make each language optional some time in the future. I just can't tell you when exactly.

I think the library will still work as long as you don't try to access any of the removed languages. For this matter, you should also remove the respective languages from the `Language` enum. Otherwise, I'm afraid, the code will be buggy.

> Thanks in advance for the help.

You are welcome. (-:

Commit:
Remove references to dropped languages

This fully drops the references to now dropped languages.
Keep in mind tests are definitely failing but since this is to create a
custom lightweight jar it is not critical :)

Signed-off-by: Ruben Aguiar <ruben@triggerise.org>
